packages/custom-webpack/src/utils.ts

```diff
@@ -1,15 +1,24 @@
+import { resolve } from 'path';
+
 /**
- * check for TS node registration
- * @param file: file name or file directory are allowed
- * @todo tsNodeRegistration: require ts-node if file extension is TypeScript
+ * Register `ts-node` if webpack config file is a TypeScript file.
+ *
+ * @param file - webpack config file path
+ * @param root - path to project root
  */
-export function tsNodeRegister(file: string = '') {
-  if (file && file.endsWith('.ts')) {
-    // Register TS compiler lazily
-    require('ts-node').register({
-      compilerOptions: {
-        module: 'commonjs',
-      },
-    });
+export function tsNodeRegister(file = '', root = process.cwd()) {
+  if (!file.endsWith('.ts')) {
+    return;
   }
+
+  const tsConfigPath = resolve(root, 'tsconfig.app.json');
```

Comment:
Actually `tsConfig` path should be taken from the config as well (we cannot assume it's always `tsconfig.app.json`).
https://github.com/just-jeb/angular-builders/blob/master/packages/custom-webpack/examples/full-cycle-app/angular.json#L28

Reply:
Can you have a look again. I have also fixed some typing issues I encountered while moving things.